client/rest/test/plugins/metadata/metadataRoutes_spec.js

```diff
@@ -255,28 +256,48 @@ describe('metadata routes', () => {
 		});
 
 		describe('get by metal id', () => {
```

Comment:
split this up into two tests 
* one that does a single call [result of call from DB needs to be checked]
* one that does two calls [only second call, from cache, needs to be checked]

current test is checked too many things, imo

Reply:
what do you think?
```js
describe('get by metal id', () => {
	const callRouteAndAssert = async (route, req, shouldCallDb) => {
		await mockServer.callRoute(route, req).then(() => {
			expect(mockServer.res.write.calledOnce).to.equal(true);
			expect(mockServer.next.calledOnce).to.equal(true);
			expect(mockServer.res.setHeader.calledWithExactly('content-type', 'image/png')).to.equal(true);
			expect(mockServer.res.setHeader
				.calledWithExactly('Content-Disposition', 'attachment; filename="image.png"')).to.equal(true);
			expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_text')).to.equal(true);
			expect(mockServer.res.setHeader.calledWithExactly('Content-MetalText', 'db_fail_text')).to.equal(false);
			expect(db.binDataByMetalId.calledOnce).to.equal(shouldCallDb);
			if (shouldCallDb)
				expect(db.binDataByMetalId.alwaysCalledWith('metal_id')).to.equal(true);
		});
	};

	// Arrange:
	const route = mockServer.getRoute('/metadata/metal/:metalId').get();
	const req = {
		params: {
			metalId: 'metal_id',
			mimeType: 'image/png',
			fileName: 'image.png',
			download: 'true'
		}
	};

	it('returns page with results and uses DB on single call', async () => {
		// Act & Assert:
		callRouteAndAssert(route, req, true);
	});

	it('returns page with results and uses cache on second call', async () => {
		// Act:
		// First call
		await mockServer.callRoute(route, req);

		// Reset the spies
		mockServer.res.write.resetHistory();
		mockServer.next.resetHistory();
		mockServer.res.setHeader.resetHistory();
		db.binDataByMetalId.resetHistory();

		// Assert:
		await callRouteAndAssert(route, req, false);
	});
});
```